### PR TITLE
[SDAN-52] - Validate the email of signed up user

### PR DESCRIPTION
**Pull request by akintolga**
- An email will be sent for new users to validate email address
- Users can ask for validation email again
- Login will fail for not-yet-validated users
- Login will fail if user's company is disabled [SDAN-49]
- Administrators can resend a validation email again [SDAN-43]
- Administrators can send a reset password token [SDAN-42]

### CODE REVIEW

**Comment by akintolga** [comment]
Depends on https://github.com/superdesk/superdesk-core/pull/1027

File: newsroom/auth/views.py [hunk 1]
```diff
@@ -15,7 +18,16 @@ def login():
         user = get_resource_service('auth_user').find_one(req=None, email=form.email.data)
         if user is not None and _is_password_valid(form.password.data.encode('UTF-8'), user):
             user = get_resource_service('users').find_one(req=None, _id=user['_id'])
-            if is_account_valid(user):
+
+            if not user.get('is_validated'):
+                flask.flash('Your email address needs validation.', 'danger')
```
Comment (petrjasek) [inline newsroom/auth/views.py]: these flash messages can use gettext

File: newsroom/auth/views.py [hunk 2]
```diff
@@ -87,3 +116,61 @@ def _modify_user_data(new_user):
...
+
+def _add_token_data(user):
+    user['token'] = get_random_string()
+    user['token_expiry_date'] = utcnow() + timedelta(days=app.config['VALIDATE_ACCOUNT_TOKEN_TIME_TO_LIVE'])
+
+
+@blueprint.route('/validate')
+def validate_account():
+    token = flask.request.args['token']
+    if token:
```
Comment (petrjasek) [inline newsroom/auth/views.py]: maybe this route should be `/validate/<token>` so there is no need for this `if token:`

File: newsroom/email.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    :param subject: Subject text
+    :param text_body: Text Body
+    :param html_body: Html Body
+    :return:
+    """
+    msg = Message(subject=subject, sender=current_app.config['MAIL_DEFAULT_SENDER'], recipients=to)
+    msg.body = text_body
+    msg.html = html_body
+    app = current_app._get_current_object()
+    thr = Thread(target=send_async_email, args=[app, msg])
```
Comment (petrjasek) [inline newsroom/email.py]: imo this is premature optimization :wink:

**Comment by akintolga** [comment]
I think for the templates we may need to route to a different folder or filename. So if the local language is 'es' then we could have `/es/account_created.html` or `account_created.html.es` which would make easier to maintain.

**Pull request merged**